src/EventListener/MergeHttpHeadersListener.php

```diff
@@ -150,7 +150,7 @@ private function mergeHttpHeaders(Response $response)
     private function getHeaders()
     {
         if (null === $this->headers) {
-            return headers_list();
+            $this->headers = headers_list();
```

Comment:
This is wrong! Exactly the reason why we had to change the listener. The headers can change between (sub)requests, that's why we have to fetch them every time.